app/amity.py

```diff
@@ -0,0 +1,328 @@
+                    print 'Room is not available'
+                    return
+            except IndexError:
+                print('No rooms available')
+                return
+            # assign an office to a person
+            # if randomized_office.is_not_full:
+                # adding a person to the member list in office
+            randomized_office.members.append(person)
+            if wants_accomodation == 'Y':
+                # pick a random living space from living space lists
+                try:
+                    random_livingspace = random.choice(self.livingspaces)
+                    room_status = self.is_room_full(random_livingspace)
+                    if room_status:
+                        print 'Room is not available'
+                        return
+                except IndexError:
+                    return 'No livingspace available'
+
+                if random_livingspace.is_not_full:
+                    # assign living space to a person
+                    person.hostel = random_livingspace
+                    # add a person to the member list in office
+                    random_livingspace.members.append(person)
+            # add the person to fellows list
+            self.fellows.append(person)
+            # add the person to all rooms
+            self.all_people.append(person)
+
+        elif role == 'staff':
+            person = Staff(name)
+            # pick a random office
+            try:
+                randomized_office = random.choice(self.office_rooms)
+                room_status = self.is_room_full(randomized_office)
+                if room_status:
+                    print 'Room is not available'
+                    return
+            except IndexError:
+                print('No rooms available')
+                return
+            if randomized_office.is_not_full is True:
+                # assign an office to a person
+                person.office = randomized_office
+                # adding a person to the member list in staff
+                randomized_office.members.append(person)
+
+            # assign the person to the staff list
+            self.staff.append(person)
+            # add staff to all people
+            self.all_people.append(person)
+        else:
+            return 'Invalid Role'
+
+    def is_room_full(self, random_room):
+        '''this method checks a room is full'''
+        if isinstance(random_room, Office):
+            if len(random_room.members) == random_room.capacity:
+                return True
+        if isinstance(random_room, LivingSpace):
+            if len(random_room.members) == random_room.capacity:
+                return True
+
+    def reallocate_person(self, name, new_room):
+        # go through the list of all people
+        for people in self.all_people:
+            # checking if that person exists
+            if name == people.name:
+                person_found = people
+                break
```

Comment:
`person_found = [person for person in self.all_people if person.name == name][0]`